src/RedirectSubscriber.php

```diff
@@ -69,17 +93,62 @@ public static function getSubscribedEvents() {
    *   The event.
    */
   public function checkForRedirection(GetResponseEvent $event) {
-    if (!$this->gdprConsentManager->needConsent()) {
+    $route_name = $this->routeMatch->getRouteName();
+
+    if ($route_name == 'gdpr_consent.data_policy.agreement') {
+      return;
+    }
+
+    if ($this->currentUser->hasPermission('without consent')) {
+      return;
+    }
+
+    $config = $this->configFactory->get('gdpr_consent.data_policy');
+
+    $entity_id = $config->get('entity_id');
+
+    /** @var \Drupal\gdpr_consent\DataPolicyStorageInterface $data_policy_storage */
+    $data_policy_storage = $this->entityTypeManager->getStorage('data_policy');
+
+    /** @var \Drupal\gdpr_consent\Entity\DataPolicyInterface $data_policy */
+    $data_policy = $data_policy_storage->load($entity_id);
+
+    $vids = $data_policy_storage->revisionIds($data_policy);
+
+    $vid = end($vids);
+
+    $values = [
+      'user_id' => $this->currentUser->id(),
+      'data_policy_revision_id' => $vid,
+    ];
+
+    if ($enforce_consent = !empty($config->get('enforce_consent'))) {
+      $values['status'] = TRUE;
+    }
+
+    $user_consents = $this->entityTypeManager->getStorage('user_consent')
+      ->loadByProperties($values);
+
+    if (!empty($user_consents)) {
+      return;
+    }
+
+    if (!$enforce_consent) {
+      $link = Link::createFromRoute(t('here'), 'gdpr_consent.data_policy.agreement');
+
+      \Drupal::messenger()->addStatus(t('You can agree with the data policy @url.', [
```

Comment:
We should this text a little bit, maybe something like:

> We published a new revision of the data policy. You can review the data policy here.

Reply:
@bramtenhove, can I use word **version** instead of **revision**? I think it will be more clear for users.